### ISSUE TITLE
Some attestations fail to include in chain

### ISSUE THREAD

**Issue by stefa2k** [prysmaticlabs/prysm#6673]
# 🐞 Bug Report

### Description

Some attestations fail even if logs indicate that attestation got submitted. Looks like this happens for all my accounts.

**Steps I consider, will report later:**
1) disable external slasher protection on service validator https://github.com/stefa2k/prysm-docker-compose/blob/5f025af7d18dfaab599fc09f0cf569ff6bb8829a/config/ha/validator-wallet.yaml#L15 -> **didn't work, also loosing attestations**
2) disable local detection on service validator https://github.com/stefa2k/prysm-docker-compose/blob/5f025af7d18dfaab599fc09f0cf569ff6bb8829a/config/ha/validator-wallet.yaml#L13 -> begin test at 2020-07-22 9:00 UTC -> ~~**seems to work, no more missing attestations**~~ **doesn't work**
3) disable slasher completely -> **not working**

### Has this worked before in a previous version?

Yes, worked with docker tag `HEAD-e0c803` and previous versions

## 🔥 Error

Unfortunately no errors.

Validator log:
```
time="2020-07-22 06:39:56" level=info msg="Submitted new attestations" AggregatorIndices=[] AttesterIndices=[10584 10174 10399] BeaconBlockRoot=0xa9ff679b29af CommitteeIndex=1 Slot=274012 SourceEpoch=8561 SourceRoot=0x0a999e6d763b TargetEpoch=8562 TargetRoot=0xaf80784bce8b prefix=validator
```

https://beaconscan.com/onyx/slot/274012#attestations but 10174 still missed the attestation.

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
Ubuntu 20.04
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
docker tag `HEAD-367738`
  </code>
</pre>

**Anything else relevant (validator index / public key)?**
index 10174
public key 0xb00411b66e7dc0a5ded34607dbf4a9cffca78d05169f600a3b0c1a590fb182888e0f742e74939ad376a2b1a1c66e7093

**Comment by xuyenvuong** [comment]
Additional logs

Validator logs
`time="2020-07-22 13:07:20" level=error msg="Could not submit slot signature to beacon node: rpc error: code = Internal desc = Could not find attestation for slot and committee in pool" prefix=validator slot=275949
time="2020-07-22 13:07:24" level=error msg="Could not submit attestation to beacon node" error="rpc error: code = DeadlineExceeded desc = context deadline exceeded" prefix=validator pubKey=0x978eedd06abf slot=275949
time="2020-07-22 13:50:12" level=error msg="Could not submit attestation to beacon node" error="rpc error: code = DeadlineExceeded desc = context deadline exceeded" prefix=validator pubKey=0xb4b43779ba40 slot=276163
time="2020-07-22 13:50:24" level=error msg="Could not submit attestation to beacon node" error="rpc error: code = DeadlineExceeded desc = context deadline exceeded" prefix=validator pubKey=0xa3c5e99b84a4 slot=276164
time="2020-07-22 14:29:36" level=error msg="Could not submit attestation to beacon node" error="rpc error: code = DeadlineExceeded desc = context deadline exceeded" prefix=validator pubKey=0xae6dcd9c9665 slot=276360
time="2020-07-22 14:29:48" level=error msg="Could not submit attestation to beacon node" error="rpc error: code = DeadlineExceeded desc = context deadline exceeded" prefix=validator pubKey=0xa05314f912ec slot=276361
time="2020-07-22 14:29:56" level=error msg="Could not submit slot signature to beacon node: rpc error: code = Internal desc = Could not find attestation for slot and committee in pool" prefix=validator slot=276362
time="2020-07-22 14:38:24" level=error msg="Could not submit attestation to beacon node" error="rpc error: code = DeadlineExceeded desc = context deadline exceeded" prefix=validator pubKey=0xb7797e85b719 slot=276404
time="2020-07-22 14:38:36" level=error msg="Could not submit attestation to beacon node" error="rpc error: code = DeadlineExceeded desc = context deadline exceeded" prefix=validator pubKey=0xb4b43779ba40 slot=276405
time="2020-07-22 14:38:36" level=error msg="Could not submit attestation to beacon node" error="rpc error: code = DeadlineExceeded desc = context deadline exceeded" prefix=validator pubKey=0xa3c5e99b84a4 slot=276405
time="2020-07-22 14:51:48" level=error msg="Could not submit attestation to beacon node" error="rpc error: code = DeadlineExceeded desc = context deadline exceeded" prefix=validator pubKey=0xb7797e85b719 slot=276471`

Slasher logs
`time="2020-07-22 00:51:32" level=error msg="rpc error: code = Internal desc = Could not insert proposer slashing │ethereum@ethnode-d815b1b5f:~$ 
into pool: slashing object already exists in pending proposer slashings" prefix=beaconclient                     │ethereum@ethnode-d815b1b5f:~$ 
time="2020-07-22 01:01:42" level=error msg="rpc error: code = Internal desc = Could not insert proposer slashing │ethereum@ethnode-d815b1b5f:~$ 
into pool: could not verify proposer slashing: validator with key 0xaa8c6a83a494b6e95523d3110f41875a3a5aee2eeeeb0│ethereum@ethnode-d815b1b5f:~$ 
185d322eec96eba4e082ec64b6c6a3a039f6cc84a4c23510b45 is not slashable" prefix=beaconclient                        │ethereum@ethnode-d815b1b5f:~$ 
time="2020-07-22 16:48:44" level=error msg="Could not receive attestations from beacon node. rpc status: Internal│ethereum@ethnode-d815b1b5f:~$ 
" error="rpc error: code = Internal desc = Could not retrieve committees for target root 0xd7e394faf094bc76500e51│ethereum@ethnode-d815b1b5f:~$ 
29b8a19c284daf18c6e26071b24f9c8740a8c12bc6: rpc error: code = Internal desc = Could not get state: could not get │ethereum@ethnode-d815b1b5f:~$ 
last saved block: highest saved block does not equal to 1, it equals to 2" prefix=beaconclient`

**Comment by handelaar2** [comment]
I think I have the same issue: validator log deadline exceeded.
`[2020-07-26 13:44:41] DEBUG gRPC stream started. backend=[] method=/ethereum.eth.v1alpha1.BeaconNodeValidator/WaitForChainStart
[2020-07-26 13:44:41]  INFO validator: Waiting for beacon chain start log from the ETH 1.0 deposit contract
[2020-07-26 13:44:41]  INFO validator: Beacon chain started genesisTime=2020-06-14 05:17:24 +0000 UTC
[2020-07-26 13:44:41] DEBUG gRPC request finished. backend=[] duration=1.467134ms method=/ethereum.eth.v1alpha1.Node/GetSyncStatus
[2020-07-26 13:44:41] DEBUG gRPC stream started. backend=[] method=/ethereum.eth.v1alpha1.BeaconNodeValidator/WaitForActivation
[2020-07-26 13:44:41]  INFO validator: Validator activated index=5577 publicKey=0xb7b72b48363c
[2020-07-26 13:44:41]  INFO validator: Validator activated index=5570 publicKey=0xa70aa5bc027e
[2020-07-26 13:44:41]  INFO validator: Validator activated index=5571 publicKey=0x83855c440a1d
[2020-07-26 13:44:41]  INFO validator: Validator activated index=5573 publicKey=0xa17fb39f69ec
[2020-07-26 13:44:41]  INFO validator: Validator activated index=5575 publicKey=0xb110fb2b25a1
[2020-07-26 13:44:41]  INFO validator: Validator activated index=5579 publicKey=0x8ebaded2bce4
[2020-07-26 13:44:41]  INFO validator: Validator activated index=5582 publicKey=0xb1dd43b5cdb2
[2020-07-26 13:44:41]  INFO validator: Validator activated index=5583 publicKey=0xa464ab7dd305
[2020-07-26 13:44:41]  INFO validator: Validator activated index=5546 publicKey=0xaee4942b0eb4
[2020-07-26 13:44:41]  INFO validator: Validator activated index=5572 publicKey=0x8c4208c9b5db
[2020-07-26 13:44:41] DEBUG gRPC request finished. backend=[] duration=5.706772ms method=/ethereum.eth.v1alpha1.BeaconChain/GetChainHead
[2020-07-26 13:44:41] DEBUG gRPC request finished. backend=[] duration=21.31059ms method=/ethereum.eth.v1alpha1.BeaconNodeValidator/GetDuties
[2020-07-26 13:44:41]  INFO validator: Attestation schedule attesters=1 pubKeys=[0x8ebaded2bce4] slot=304930
[2020-07-26 13:44:41]  INFO validator: Attestation schedule attesters=1 pubKeys=[0xaee4942b0eb4] slot=304931
[2020-07-26 13:44:41]  INFO validator: Attestation schedule attesters=1 pubKeys=[0xa464ab7dd305] slot=304934
[2020-07-26 13:44:41]  INFO validator: Attestation schedule attesters=1 pubKeys=[0xa17fb39f69ec] slot=304936
[2020-07-26 13:44:41]  INFO validator: Attestation schedule attesters=1 pubKeys=[0x8c4208c9b5db] slot=304937
[2020-07-26 13:44:41]  INFO validator: Attestation schedule attesters=2 pubKeys=[0xb1dd43b5cdb2 0xb110fb2b25a1] slot=304946
[2020-07-26 13:44:41]  INFO validator: Attestation schedule attesters=2 pubKeys=[0x83855c440a1d 0xb7b72b48363c] slot=304953
[2020-07-26 13:44:41]  INFO validator: Attestation schedule attesters=1 pubKeys=[0xa70aa5bc027e] slot=304956
[2020-07-26 13:44:41] DEBUG gRPC request finished. backend=[] duration=3.997745ms method=/ethereum.eth.v1alpha1.BeaconNodeValidator/DomainData
[2020-07-26 13:44:41] DEBUG gRPC request finished. backend=[] duration=141.729167ms method=/ethereum.eth.v1alpha1.BeaconNodeValidator/GetDuties
[2020-07-26 13:44:41] DEBUG gRPC request finished. backend=[] duration=1.288263ms method=/ethereum.eth.v1alpha1.BeaconNodeValidator/DomainData
[2020-07-26 13:44:41] DEBUG gRPC request finished. backend=[] duration=1.540146ms method=/ethereum.eth.v1alpha1.BeaconNodeValidator/SubscribeCommitteeSubnets
[2020-07-26 13:44:51] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:43:48 +0000 UTC slot=304931
[2020-07-26 13:44:55] DEBUG gRPC request finished. backend=[] duration=111.584µs method=/ethereum.eth.v1alpha1.BeaconNodeValidator/GetAttestationData
[2020-07-26 13:44:55] ERROR validator: Could not request attestation to sign at slot error=rpc error: code = DeadlineExceeded desc = context deadline exceeded pubKey=0xaee4942b0eb4 slot=304931
[2020-07-26 13:45:03] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:44:00 +0000 UTC slot=304932
[2020-07-26 13:45:15] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:44:12 +0000 UTC slot=304933
[2020-07-26 13:45:27] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:44:24 +0000 UTC slot=304934
[2020-07-26 13:45:31] DEBUG gRPC request finished. backend=[] duration=56.723µs method=/ethereum.eth.v1alpha1.BeaconNodeValidator/GetAttestationData
[2020-07-26 13:45:31] ERROR validator: Could not request attestation to sign at slot error=rpc error: code = DeadlineExceeded desc = context deadline exceeded pubKey=0xa464ab7dd305 slot=304934
[2020-07-26 13:45:39] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:44:36 +0000 UTC slot=304935
[2020-07-26 13:45:51] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:44:48 +0000 UTC slot=304936
[2020-07-26 13:45:55] DEBUG gRPC request finished. backend=[] duration=245.359µs method=/ethereum.eth.v1alpha1.BeaconNodeValidator/GetAttestationData
[2020-07-26 13:45:55] ERROR validator: Could not request attestation to sign at slot error=rpc error: code = DeadlineExceeded desc = context deadline exceeded pubKey=0xa17fb39f69ec slot=304936
[2020-07-26 13:46:03] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:45:00 +0000 UTC slot=304937
[2020-07-26 13:46:07] DEBUG gRPC request finished. backend=[] duration=52.538µs method=/ethereum.eth.v1alpha1.BeaconNodeValidator/GetAttestationData
[2020-07-26 13:46:07] ERROR validator: Could not request attestation to sign at slot error=rpc error: code = DeadlineExceeded desc = context deadline exceeded pubKey=0x8c4208c9b5db slot=304937
[2020-07-26 13:46:15] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:45:12 +0000 UTC slot=304938
[2020-07-26 13:46:27] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:45:24 +0000 UTC slot=304939
[2020-07-26 13:46:39] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:45:36 +0000 UTC slot=304940
[2020-07-26 13:46:51] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:45:48 +0000 UTC slot=304941
[2020-07-26 13:47:03] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:46:00 +0000 UTC slot=304942
[2020-07-26 13:47:15] DEBUG validator: Set deadline for proposals and attestations deadline=2020-07-26 13:46:12 +0000 UTC slot=304943`
Running latest beacon and validator. Beacon runs with --dev

**Comment by stefa2k** [comment]
happened again with validator no. 2107:
```
    time="2020-08-04 16:44:12" level=info msg="Submitted new attestations" AggregatorIndices=[] AttesterIndices=[2425 2623 2817 2835 2356 2944 2107 2097 2899] BeaconBlockRoot=0xa43745a65167 CommitteeIndex=0 Slot=1120 SourceEpoch=34 SourceRoot=0x65ec48482607 TargetEpoch=35 TargetRoot=0xa43745a65167 prefix=validator
```

but shown as missed: https://beaconcha.in/validator/945d36ce4fcfc03b056ad950e9fb330ea107afc83789241dd7dfcc515d7ee9a9110477b3596a7842681a62a43eb98d4c#attestations

**Comment by terencechain** [comment]
It is included. I found the attestation by 2107. It's included in block 1122 
See:  https://beaconcha.in/block/1122#pills-attestations

![Screen Shot 2020-08-04 at 10 14 03 AM](https://user-images.githubusercontent.com/21316537/89323867-3a6f2b00-d63b-11ea-83e7-5927517cfda6.png)

It looks more like a status display bug on `https://beaconcha.in/validator/`

**Comment by terencechain** [comment]
> happened again with validator no. 2107:
> 
> ```
>     time="2020-08-04 16:44:12" level=info msg="Submitted new attestations" AggregatorIndices=[] AttesterIndices=[2425 2623 2817 2835 2356 2944 2107 2097 2899] BeaconBlockRoot=0xa43745a65167 CommitteeIndex=0 Slot=1120 SourceEpoch=34 SourceRoot=0x65ec48482607 TargetEpoch=35 TargetRoot=0xa43745a65167 prefix=validator
> ```
> 
> but shown as missed: https://beaconcha.in/validator/945d36ce4fcfc03b056ad950e9fb330ea107afc83789241dd7dfcc515d7ee9a9110477b3596a7842681a62a43eb98d4c#attestations

It shows as attested now:
https://beaconcha.in/validator/945d36ce4fcfc03b056ad950e9fb330ea107afc83789241dd7dfcc515d7ee9a9110477b3596a7842681a62a43eb98d4c#attestations

Seems like the attested status is lagged

**Comment by stefa2k** [comment]
Thanks, yes, I could verify it with 2 more. Looks like its a bug in beaconcha.in.

**Comment by stefa2k** [comment]
Happened again, will have an eye on it but I think its simply a display bug of beaconcha.in

Thanks!

**Comment by stefa2k** [comment]
Looks like its still there:
https://beaconcha.in/validator/b1c8c3ac9508bb170200b25b7222b3f985e7bdfe86584e77ca121b9e2709e285a60f32769cffa6dcde18920bdf96ddcd#attestations (validator 2085)
Slot 3844: https://beaconscan.com/slot/3844#votes
log:
```
time="2020-08-05 01:49:00" level=info msg="Submitted new attestations" AggregatorIndices=[] AttesterIndices=[2085 2805 2751 2803 2430] BeaconBlockRoot=0xb33d1a8783b0 CommitteeIndex=1 Slot=3844 SourceEpoch=119 SourceRoot=0x51c75628a73c TargetEpoch=120 TargetRoot=0xa25ca59b37af prefix=validator
```

**Comment by terencechain** [comment]
I'll write a debug end point to better understand this problem

**Comment by terencechain** [comment]
> Looks like its still there:
> https://beaconcha.in/validator/b1c8c3ac9508bb170200b25b7222b3f985e7bdfe86584e77ca121b9e2709e285a60f32769cffa6dcde18920bdf96ddcd#attestations (validator 2085)
> Slot 3844: https://beaconscan.com/slot/3844#votes
> log:
> 
> ```
> time="2020-08-05 01:49:00" level=info msg="Submitted new attestations" AggregatorIndices=[] AttesterIndices=[2085 2805 2751 2803 2430] BeaconBlockRoot=0xb33d1a8783b0 CommitteeIndex=1 Slot=3844 SourceEpoch=119 SourceRoot=0x51c75628a73c TargetEpoch=120 TargetRoot=0xa25ca59b37af prefix=validator
> ```

Updates:
 * I added a debug endpoint in #6895 for observatory. I can confirm that all your missed attestations were never included in the chain. Hence this is not a block chain explorer's issue, this is a networking and/or protocol issue.
 * @prestonvanloon has been working on an improvement fix for this in #6893, we are confident that will solve this issue

**Comment by rauljordan** [comment]
Hi @stefa2k we have since then significantly improved this code path. Let us know if this issue still persists and we will reopen as needed
